### PR TITLE
build: upgrade TypeScript to 2.6

### PR DESCRIPTION
**Pull request by gkalpak**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: packages/service-worker/worker/src/service-worker.d.ts [hunk 1]
```diff
@@ -25,32 +25,33 @@ interface ExtendableEvent extends Event {
...
-  open(cacheName: string): Promise<Cache>;
-}
-
-interface CacheStorageOptions {
-  cacheName?: string;
-  ignoreMethod?: boolean;
-  ignoreSearch?: boolean;
-  ignoreVary?: boolean;
-}
+// TypeScript 2.6's `lib.dom.d.ts` includes improved definitions for these.
```
Comment (IgorMinar) [inline packages/service-worker/worker/src/service-worker.d.ts]: we should just delete these.

**Review by IgorMinar — COMMENTED** [review]
this is shaping up nicely. the CI is unhappy though so more work is needed.

Chuck has also more changes that @mprobst put together within g3 to make our code base compile.

I'll leave it to you and @chuckjaz to sync up on what needs to be done to finish this work.

**Comment by gkalpak** [comment]
The current Travis failures are due to Tsickle not being compatible with TypeScript 2.6 yet.

**Comment by IgorMinar** [comment]
one of Martin, Chuck and AlexE are working on fixing tsickle. Chuck has all
the details.

**Comment by gkalpak** [comment]
@chuckjaz, should this be closed in favor of #21033?

**Pull request closed without merging**